backend/service/feedback/feedback.go

```diff
@@ -0,0 +1,132 @@
+	p, ok := service.Registry[pgservice.Name]
+	if !ok {
+		return nil, fmt.Errorf("could not find the %v database service", pgservice.Name)
+	}
+
+	dbClient, ok := p.(pgservice.Client)
+	if !ok {
+		return nil, errors.New("database does not implement the required interface")
+	}
+
+	return &svc{storage: &storage{db: dbClient.DB()}, logger: logger, scope: scope}, nil
+}
+
+type svc struct {
+	storage *storage
+	logger  *zap.Logger
+	scope   tally.Scope
+}
+
+type storage struct {
+	db *sql.DB
+}
+
+type submission struct {
+	id          string
+	submittedAt time.Time
+	userId      string
+	score       int64
+	feedback    *feedbackv1.Feedback
+	metadata    *feedbackv1.FeedbackMetadata
+}
+
+// used to normalize a given emoji rating out of a 100
+// can add more emoji values as new use cases arise
+var emojiRatingScore = map[string]int64{
+	feedbackv1.EmojiRating_SAD.String():     30,
+	feedbackv1.EmojiRating_NEUTRAL.String(): 70,
+	feedbackv1.EmojiRating_HAPPY.String():   100,
+}
+
+type Service interface {
+	SubmitFeedback(ctx context.Context, id string, userId string, feedback *feedbackv1.Feedback, metadata *feedbackv1.FeedbackMetadata) error
+}
+
+func (s *svc) processSubmission(id string, userId string, feedback *feedbackv1.Feedback, metadata *feedbackv1.FeedbackMetadata) (*submission, error) {
+	if len(id) < 36 {
+		return nil, status.Error(codes.InvalidArgument, "client id was not a uuid length")
+	}
+
+	if userId == "" {
+		return nil, status.Error(codes.InvalidArgument, "user id was empty")
+	}
+
+	if feedback == nil || metadata == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "feedback: %v or metadata: %v provided was nil", feedback, metadata)
+	}
+
+	score, err := calculateRatingScore(feedback.RatingScale)
+	if err != nil {
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
+	}
+
+	// the main question that was asked in the feedback component
+	if metadata.Survey.Prompt == "" {
+		return nil, status.Error(codes.InvalidArgument, "metadata survey prompt was empty")
+	}
+
+	if metadata.Survey.RatingLabels == nil {
+		return nil, status.Error(codes.InvalidArgument, "metadata rating options was nil")
+	}
+
+	return &submission{
+		id:          id,
+		submittedAt: time.Now(),
+		userId:      userId,
+		score:       score,
+		feedback:    feedback,
+		metadata:    metadata,
+	}, nil
+}
+
+// looks up the enum's rating score using its corresponding rating scale
+func calculateRatingScore(scale *feedbackv1.RatingScale) (int64, error) {
+	switch scale.Type.(type) {
+	case *feedbackv1.RatingScale_Emoji:
+		rating := scale.GetEmoji().String()
+		v, ok := emojiRatingScore[scale.GetEmoji().String()]
+		if !ok {
+			return -1, fmt.Errorf("unsupported rating: %v", rating)
+		}
+		return v, nil
+	default:
+		return -1, fmt.Errorf("unsupported rating scale: %v", scale)
+	}
+}
+
+func (s *svc) SubmitFeedback(ctx context.Context, id string, userId string, feedback *feedbackv1.Feedback, metadata *feedbackv1.FeedbackMetadata) error {
```

Comment:
I could see the `uuid` being an optional param if we're going to support upserting, if we decided to implement feedback checkpointing. But until then I dont think its best for the caller to have to generate the `uuid`. 

If we do support feedback checkpointing, Submit feedback should return a `uuid` upon first submission and if one is provided then we would do the upsert.

But for now, why have the caller be required to generate a `uuid`?

Reply:
this was a suggestion/decision made in a previous PR https://github.com/lyft/clutch/pull/1828 
the advantage of this being generated by the client is that this approach doesn't require coordination from the server. also it's required, because for the V1, we are sending a feedback request each time the user selects an emoji and updating the feedback at the id with the latest submission

Reply:
I see your upsert query below, we can stick to the original decision.